lib/ui/platform_dispatcher.dart

```diff
@@ -1299,6 +1303,102 @@ class PlatformDispatcher {
 
   @Native<Handle Function()>(symbol: 'PlatformConfigurationNativeApi::DefaultRouteName')
   external static String _defaultRouteName();
+
+  /// Scales the `unscaledFontSize`, according to the user's platform
+  /// preferences.
+  ///
+  /// Many platforms allow users to scale text globally for better readability.
+  /// Given the font size the app developer specified in logical pixels, this
+  /// method converts it to the preferred font size (also in logical pixels) that
+  /// accounts for platform-wide text scaling.
+  ///
+  /// Returns null if the platform only supports linear text scaling, or does
+  /// not support platform-wide text scaling. The return value is always
+  /// non-negative and finite if not null.
+  ///
+  /// The scaled value of the same font size input may change if the user changes
+  /// the text scaling preference (in system settings for example). The
+  /// [onTextScaleFactorChanged] callback can be used to monitor such changes.
+  ///
+  /// Instead of directly calling this method, applications should typically use
+  /// [MediaQuery.textScalerOf] to retrive the scaled font size in a widget tree,
+  /// so text in the app resizes properly when the text scaling preference
+  /// changes.
+  double? scaleFontSize(double unscaledFontSize) {
+    assert(unscaledFontSize >= 0);
+    assert(unscaledFontSize.isFinite);
+
+    if (textScaleFactor == 1.0) {
+      return unscaledFontSize;
+    }
+
+    final int unscaledFloor = unscaledFontSize.floor();
+    final int unscaledCeil = unscaledFontSize.ceil();
+    if (unscaledFloor == unscaledCeil) {
+      return _scaleAndMemoize(unscaledFloor);
+    }
+    assert(unscaledCeil - unscaledFloor == 1);
+
+    return switch ((_scaleAndMemoize(unscaledFloor), _scaleAndMemoize(unscaledCeil))) {
+      (null, _) => null,
+      (_, null) => null,
+      (final double lower, final double upper) => lower + (upper - lower) * (unscaledFontSize - unscaledFloor),
+    };
+  }
+
+  // The _cachedFontSizes variable is set to null if GetScaledFontSize is not
+  // implemented on the current platform.
+  Map<int, double>? _cachedFontSizes = <int, double>{};
```

Comment:
I just changed the return type of the public method to `double` so the framework doesn't know whether the scaling curve is linear (in which case there's no need to do caching) so this can only happen in the engine now. 

> Why do we cache in the first place? Are these calculations sooo expensive?

I can't see the implementation so I can't really answer that. I'll get someone that has the answer.